matryoshka_tester/helpers.py

```diff
@@ -0,0 +1,142 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from os import getenv
+from typing import Any, Optional
+import pytest
+
+import testinfra
+
+
+@dataclass(frozen=True)
+class ToParamMixin:
+    """
+    Mixin class that gives child classes the ability to convert themselves into
+    a pytest.param with self.__str__() as the default id and optional marks
+    """
+
+    marks: Any = None
+
+    def to_pytest_param(self):
+        return pytest.param(self, id=self.__str__(), marks=self.marks or ())
+
+
+@dataclass(frozen=True)
+class OciRuntimeBase(ABC, ToParamMixin):
+    #: command that builds the Dockerfile in the current working directory
+    build_command: str = ""
+    #: the "main" binary of this runtime, e.g. podman or docker
+    runner_binary: str = ""
+    _runtime_functional: bool = False
+
+    def __post_init__(self) -> None:
+        if not self.build_command or not self.runner_binary:
+            raise ValueError(
+                f"build_command ({self.build_command=}) or runner_binary "
+                f"({self.runner_binary=}) were not specified"
+            )
+        if not self._runtime_functional:
+            raise RuntimeError(
+                f"The runtime {self.__class__.__name__} is not functional!"
+            )
+
+    @abstractmethod
+    def get_image_id_from_stdout(self, stdout: str) -> str:
+        pass
+
+    def __str__(self) -> str:
+        return self.__class__.__name__
+
+
+LOCALHOST = testinfra.host.get_host("local://")
+
+_DOCKER_WORKS = LOCALHOST.run("docker ps").succeeded
```

Comment:
I was thinking about doing that, but it would then run this check for **every** instantiation of the `container_runtime` which causes needles system calls.